BlueprintUI/Tests/StackTests.swift

```diff
@@ -1176,13 +1176,162 @@ class StackTests: XCTestCase {
+                    onMeasure(constraint)
+
+                    switch size {
+                    case .fixed(let size):
+                        return size
+
+                    case .area(let total):
+                        switch constraint.width {
+                        case .atMost(let max):
+                            let width = min(total, max)
+                            return CGSize(width: width, height: round(total / width))
+                        case .unconstrained:
+                            return CGSize(width: total, height: 1)
+                        }
+
+                    }
+                }
+            }
+
+            func backingViewDescription(with context: ViewDescriptionContext) -> ViewDescription? {
+                nil
+            }
+        }
+
+        var fixed1 = [SizeConstraint]()
+        var fixed2 = [SizeConstraint]()
+        var fixed3 = [SizeConstraint]()
+        var flexible1 = [SizeConstraint]()
+
+        let row = Row(alignment: .fill, underflow: .growUniformly, overflow: .condenseUniformly) {
+
+            MeasurementCounter(
+                size: .fixed(CGSize(width: 20, height: 30))
+            ) { size in
+                fixed1.append(size)
+            }
+            .stackLayoutChild(priority: .fixed)
+
+            MeasurementCounter(
+                size: .fixed(CGSize(width: 21, height: 30))
+            ) { size in
+                fixed2.append(size)
+            }
+            .stackLayoutChild(priority: .fixed)
+
+            Row(alignment: .fill, underflow: .growUniformly, overflow: .condenseUniformly) {
+
+                MeasurementCounter(
+                    size: .fixed(CGSize(width: 22, height: 30))
+                ) { size in
+                    fixed3.append(size)
+                }
+                .stackLayoutChild(priority: .fixed)
+
+                MeasurementCounter(
+                    size: .area(total: 1000)
+                ) { size in
+                    flexible1.append(size)
+                }
+                .stackLayoutChild(priority: .flexible)
+            }
+            .stackLayoutChild(priority: .flexible)
+        }
+
+        BlueprintStackUpdates.isOptimizedLayoutEnabled = true
+
+        _ = row.content.measure(in: SizeConstraint(CGSize(width: 100, height: 30)))
+
+        XCTAssertEqual(fixed1.count, 1)
+
+        XCTAssertEqual(
+            fixed1,
+            [
+                SizeConstraint(width: .atMost(100), height: .atMost(30)),
+            ]
+        )
+
+        XCTAssertEqual(fixed2.count, 1)
+
+        XCTAssertEqual(
+            fixed2,
+            [
+                SizeConstraint(width: .atMost(100), height: .atMost(30)),
+            ]
+        )
+
+        XCTAssertEqual(fixed3.count, 1)
+
+        XCTAssertEqual(
+            fixed3,
+            [
+                SizeConstraint(width: .atMost(59), height: .atMost(30)),
+            ]
+        )
+
+        XCTAssertEqual(flexible1.count, 1)
```

Comment:
what are the counts for each of these before you made these changes?

Reply:
Some were 4, and some were 2.